chatminer/chatparsers.py

```diff
@@ -236,6 +236,47 @@ def _parse_message(self, mess):
         }
         return parsed_message
 
+class InstagramChatsParser(Parser):
+    def _read_file_into_list(self):
+        self._logger.info("Starting reading raw messages into memory...")
+        self.messages = []
+        with self._file.open(encoding="utf-8") as f:
+            messages_raw = reversed((json.load(f)["messages"]))
+
+        for line in messages_raw:
+            self.messages.append(line)
+        self._logger.info(
+            "Finished reading %i raw messages into memory.", len(self.messages)
+        )
+
+    def _parse_message(self, mess):
+        if "share" in mess:
+            if "link" in mess["share"]:
+                body = mess["share"]["link"]
+            elif "share_text" in mess["share"]:
+                body = mess["share"]["share_text"]
+            else:
+                body = "{} sent {}'s story.".format(mess["sender_name"], mess["share"]["original_content_owner"])
+        elif "photos" in mess:
+            body = mess["photos"][0]["uri"]
+        elif "videos" in mess:
+            if len(mess["videos"]):
+                body = mess["videos"][0]["uri"]
+            else:
+                body = "Photo sent by {}".format(mess["sender_name"])
+        elif "audio_files" in mess:
+            body = mess["audio_files"][0]["uri"]
+        elif "content" in mess:
+            body = mess["content"]
+        else:
+            body = ""
```

Comment:
```suggestion
            self._logger.warning("Skipped message with unknown format: %s", mess)
            return None
```
In alignment with the other parsers, this gives the user some feedback on which messages fail the parsing